File: py/4673selfnum.py

from operator import indexOf


def selfnum(num):
    val = num 
    while num >0:
        val += (val%10)
        num //= 10

    return val
        
        
    
li = [0] * 10001
for i in range(10000):
    nonselfnum = selfnum(i)
    if nonselfnum < 10001:
        li[nonselfnum] = 1

for i in range(len(li)):
    if li[i]:
        print(i)
    
    

File: c/4673selfnum.c
#include <stdio.h>

int selfNum(int num){
    int ans = num;
    while(num>0){
        ans+=num%10;
        num/=10;
    }
    return ans;
}

int main(void){
    int array[10001];
    for(int i=1; i<=10001; i++){
        int check = selfNum(i);
        if(check < 10001) array[check] =1;
    }
    for(int i=1; i<10001; i++){
        if(array[i]!=1) printf("%d\n", i);
    }
    return 0;
}